### PR TITLE
Refactor 1: moved tpu xxx_step to backend

### PR DESCRIPTION
**Pull request by williamFalcon**

### CODE REVIEW

File: pytorch_lightning/accelerators/base_backend.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import torch
+from typing import Any
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+
+
+class Accelerator(object):
```
Comment (Borda) [inline pytorch_lightning/accelerators/base_backend.py:6]: let's make it abstract

**Comment by Borda** [comment]
why would you move it to Accelerator, it will be the same in each backend or how it would differ?

**Comment by williamFalcon** [comment]
yes. only way to make this more general. each backend handles its own hooks. then you can just call a backend directly

File: pytorch_lightning/accelerators/tpu_backend.py [hunk 2]
```diff
@@ -117,6 +118,47 @@ def tpu_train_in_process(self, tpu_core_idx: int, model: LightningModule, traine
...
+        batch = self.to_device(batch)
+        args[0] = batch
+        output = self.trainer.model.validation_step(*args)
+        return output
+
+    def test_step(self, batch, args):
+        batch = self.to_device(batch)
+        args[0] = batch
+        output = self.trainer.model.test_step(*args)
+        return output
```
Comment (Borda) [inline pytorch_lightning/accelerators/tpu_backend.py:137]: then this shall be in abstract class

**Pull request merged**